### PR TITLE
mtl: pm: D3 power flow errata

### PR DESCRIPTION
**Pull request by marcinszkudlinski**
Building of some legacy SOF platforms with specific configs fail after https://github.com/zephyrproject-rtos/zephyr/pull/47840 is applied. This is a hotfix for this

### CODE REVIEW

**Review by juimonen — DISMISSED** [review]
can't comment too much the actual code, but this fixes a tgl sof build for me. so approved.

**Comment by marc-hb** [comment]
Please add Zephyr commit `1e407cef8664618d0e454d7a46ec393a46ba3528` (with all 40 digits) to sof/west.yml and submit to SOF CI at https://github.com/thesofproject/sof, see if that fixes the build.

No need to change the remote or anything, a single-line `revision:` change is enough. Example: https://github.com/thesofproject/sof/pull/6605/files

File: soc/xtensa/intel_adsp/ace/boot.c
Comment (ceolin) [inline soc/xtensa/intel_adsp/ace/boot.c:49]: Is it really weird to see cavs namespace inside ACE

**Review by ceolin — APPROVED** [review]

**Review by juimonen — APPROVED** [review]

**Review by softwarecki — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Comment by kv2019i** [comment]
@marc-hb  This is really hard to test with the CI as this is a configuration not enabled by default for any platform. 

And @marcinszkudlinski why back as draft? 

If you plan to change the PR in any way, it would be good to add to the git commit description:
  - what is the problem fixed, e.g. "soc/xtensa/intel_adsp/common/boot.c:202: undefined reference to `pm_state_imr_restore' 
 on cavs2.5 if CONFIG_PM=y
  - what is the commit this is fixing
I know you have this information in this PR, but I think this information would be more useful in the git commit message itself for any future generations wondering around the code change and why it was made. But I'm not blocking this, as this is clearly needed and the information can be found by backtracking the github PR.

**Comment by marcinszkudlinski** [comment]
@kv2019i Just because I wanted to perform some extra testing with SOF and did not want to create another errata
looks fine - changing to open

**Comment by marcinszkudlinski** [comment]
As to comments - I don't want to dismiss approvals... so if comments are not really necessary pls proceed with merging

**Pull request merged**
